.pre-commit-config.yaml

```diff
@@ -0,0 +1,31 @@
+ci:
+  autofix_prs: false
+repos:
+# shared across repos:
+- repo: https://github.com/pre-commit/pre-commit-hooks
+  rev: v4.1.0
+  hooks:
+    - id: check-added-large-files
+    - id: check-case-conflict
+    - id: check-executables-have-shebangs
+    - id: check-merge-conflict
+    - id: check-shebang-scripts-are-executable
+    - id: check-symlinks
+    - id: check-xml
+    - id: check-yaml
+    - id: destroyed-symlinks
+    - id: end-of-file-fixer
+    - id: mixed-line-ending
+      args: ['--fix=lf']
+    - id: trailing-whitespace
+- repo: https://github.com/python-jsonschema/check-jsonschema
+  rev: 0.14.2
+  hooks:
+    - id: check-github-workflows
+- repo: https://github.com/Lucas-C/pre-commit-hooks
+  rev: v1.1.13
+  hooks:
+    - id: forbid-tabs
+    - id: remove-tabs
+# specific to dfhack:
+exclude: '^(depends/)'
```

Comment:
Would `exclude_types = [json]` do the trick?

Reply:
Supposedly, according to https://pre-commit.com/#pre-commit-configyaml---hooks, we can specify `exclude_types` on a per-hook basis. I tried adding `exclude_types = [json]` to the `*-tabs` hooks above and it did not appear to change their behavior at all. Maybe it is a responsibility of the hook to support that, though.

Honestly, I might tend towards excluding JSON files in the specific problematic folder from all hooks. In this case, `data/examples/orders/*.json` are the only files giving me trouble, and they seem to be generated and likely won't be hand-edited in the future, so style is not really important.